utils/small_map/src/entry.rs

```diff
@@ -219,6 +224,7 @@ impl<'a, K: Ord, V, const ARRAY_SIZE: usize> OccupiedEntry<'a, K, V, ARRAY_SIZE>
 
     /// Sets the value of the entry with the `OccupiedEntry`'s key,
     /// and returns the entry's old value.
+    #[must_use]
```

Comment:
Huh, actually after seeing the broken lints, I'm wondering if both `insert` and `remove`, should have a `must_use` attribute.

While the failure's aren't exactly incorrect, it feels like a negative of the API to force users to use `let _ = map.insert(value)` vs. `map.insert(value)`